TestApps/Auth/AuthTestApp/AuthViewModel.swift

```diff
@@ -102,7 +118,55 @@ class AuthViewModel: ObservableObject {
 			return
 		}
 
-		let duration = Duration.seconds(expiresAt.timeIntervalSinceNow)
-		expiresIn = duration.formatted(.time(pattern: .minuteSecond(padMinuteToLength: 2)))
+		if #available(iOS 16, *) {
+			let duration = Duration.seconds(expiresAt.timeIntervalSinceNow)
+			expiresIn = duration.formatted(.time(pattern: .minuteSecond(padMinuteToLength: 2)))
+		} else {
+			let timeInterval = expiresAt.timeIntervalSinceNow
+			let minutes = Int(timeInterval) / 60
+			let seconds = Int(timeInterval) % 60
+			expiresIn = String(format: "%02d:%02d", minutes, seconds)
+		}
+	}
+
+	func presentationAnchor(for session: ASWebAuthenticationSession) -> ASPresentationAnchor {
```

Comment:
[Question] maybe we need to provide another class which is not a `MainActor`?

Reply:
Yes, that works! thanks 👍